source/adios2/ADIOSTypes.h

```diff
@@ -11,36 +11,169 @@
 #ifndef ADIOS2_ADIOSTYPES_H_
 #define ADIOS2_ADIOSTYPES_H_
 
+/// \cond EXCLUDE_FROM_DOXYGEN
```

Comment:
It is but the idea is to separate the C headers from C++ headers, so for the C headers, also separate actual C headers from C++ wrapped C headers.  See https://github.com/ornladios/ADIOS2/wiki/File-Header-Structure-and-Includes